### PR TITLE
feat: upgraded hr_HR locale

### PR DESCRIPTION
**Pull request by fkukovec**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md)]

### 🤔 This is a ...

- [ ] New feature
- [ ] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [x] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Other (about what?)

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |     Upgraded HR locale      |

### ☑️ Self Check before Merge

⚠️ Please check all items below before review. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

### CODE REVIEW

**Comment by yoyo837** [comment]
Partly related to #23369

File: components/locale/hr_HR.tsx [hunk 1]
```diff
@@ -4,6 +4,8 @@ import TimePicker from '../time-picker/locale/hr_HR';
 import Calendar from '../calendar/locale/hr_HR';
 import { Locale } from '../locale-provider';
 
+const typeTemplate = '${label} nije valjan ${type}';
```
Comment (yoyo837) [inline components/locale/hr_HR.tsx:7]: Please disable `no-template-curly-in-string` lint error here.

Comment (fkukovec) [reply components/locale/hr_HR.tsx:8]: I did it, thanks!

**Review by yoyo837 — APPROVED** [review]

**Comment by yoyo837** [comment]
I think this can be submitted to the master branch. 🤔

**Pull request merged**